src/CLICommand/parameter.ts

```diff
@@ -0,0 +1,59 @@
+export type ParameterName = string;
+export type ParameterType = 'single-value' | 'boolean' | 'array';
+
+export interface ParameterData {
+	name: ParameterName;
+	aliases: ParameterName[];
+	description: string;
+	type?: ParameterType;
+	default?: string;
+	required?: true;
+	forbiddenConjunctionParameters?: ParameterName[];
+	requireedConjunctionParameters?: ParameterName[];
+}
+
+export class Parameter implements ParameterData {
+	private parameterData: ParameterData;
+	private static parameters: ParameterData[] = [];
+
+	constructor(public readonly name: ParameterName) {
+		this.parameterData = Parameter.get(name);
+	}
+
+	public get aliases(): ParameterName[] {
+		return this.parameterData.aliases;
+	}
+
+	public get description(): string {
+		return this.parameterData.description;
+	}
+
+	public get default(): string | undefined {
+		return this.parameterData.default;
+	}
+
+	public get type(): ParameterType {
+		return this.parameterData.type || 'single-value';
+	}
+
+	public static declare(parameter: ParameterData): void {
+		Parameter.parameters.push(parameter);
+	}
+
+	/**
+	 * @name reset
+	 * For testing pruposes only. It will just remove all parameters declaration
```

Comment:
purposes, similar to other nits

Reply:
c0f61a7a2ad689c94578a24172c72cb016135646